### ISSUE TITLE
Get rid of Tauri and move into a CLI only UI

### ISSUE THREAD

**Issue by Katajisto** [Katajisto/trueno#24]
Using Tauri was a bad idea. Lots of frustration, having to use JS packages. Dealing with build pipelines. Lots of ram usage.

I am going to get rid of it. We are going to move into a CLI only UI. Much better for teh workflow. 

@Arch-vile, any objections?

**Comment by Arch-vile** [comment]
I love the idea of pure CLI, but it will add a whole new level of design problems for the UI as CLI will pose some quite stricts constraints.

**Comment by Arch-vile** [comment]
First issue I can see how do we make collapsing/expanding and copy pasting parts of the response payload as easy as it would be with graphical UI?

Another thing is that graphical UI with good keyboard shortcuts would serve two user segments well whereas I am not sure with CLI only we get attraction from those used to graphical UIs.

**Comment by Katajisto** [comment]
Yep, I started on a simple terminal UI on saturday and put some work into it. Complexity-wise it's way better. Everything is in rust and I felt much better with using the Rust types for everything. I also feel way better doing it, since shipping software that uses 500mb of ram does not feel very nice. 

The issues that you mentioned are true, but I feel like the first issue is solvable with a response browsing thingy that lets you go row by row and also lets you collapse parts. Then you can also copy/paste these collapsible parts. It will need some work on it, but I would say it's reasonably solvable with like a few days of work in total. 

Then the part about the user segments, we need to think about what the goals are for this software. I haven't really though much about it yet. I think we could just try to be a small tool for people like us who prefer simple programs that do a job well. I don't know if I particularly want to cater to the masses and have them depend on me doing free work on maintaining JS/TS code. Also, are we trying to get a lot of people to use this in general?

**Comment by Arch-vile** [comment]
Creating a compelling software tool is likely to attract a broad user base, especially if we incorporate a graphical user interface (GUI). However, it's also worth considering a more targeted approach to cater to specific user segments.

One significant advantage of opting for a command-line interface (CLI) is that it encourages keyboard-first usage, which can have its own benefits.

I'm open to taking on the challenge of developing the software with these considerations!

**Comment by Katajisto** [comment]
I'm fine with both approaches CLI and GUI. Have you tried the improvements that I made before getting fed up with the GUI in 0.3.3? I introduced a way better editor for the scripts and the request body. I think we can discuss the GUI vs CLI approach on monday and also agree on some guidelines for how we'll develop the software. 

I asked chatgpt to write me a set of contributing guidelines based on my suggestions and thoughts so we could be on the same page about development if we are going to work on this together. What do you think about them? They assume that we would choose the GUI path and typescript would be used.

## Contributing Guidelines

### General Principles

1. **Separation of Concerns**: Business logic will predominantly reside in the Rust layer, leaving the frontend/CLI UI to focus mainly on rendering and user interaction.

2. **Simplicity First**: Go for easy and clear solutions; avoid complicating things unnecessarily. Taking on new 3rd party dependencies should be carefully considered.

3. **Developer Autonomy**: Trusted contributors have the latitude to express their individual coding perspectives and make independent decisions regarding implementation.

4. **Code Review Policy**: Code reviews will be bypassed for contributions from trusted developers, initially comprising just the project maintainers.

5. **Code Style**: While no formal style guide will be enforced, contributors outside of the trusted pool must adhere to a reasonable and clean coding style. Elevation to the trusted pool will be based on proven skill in writing maintainable code.

### Tooling

#### TypeScript

6. **Tooling - TypeScript**: No ESLint or Prettier will be used. Developers are expected to manage their own formatting.

#### Rust

7. **Tooling - Rust**: Use of `rustfmt` is optional. If you're working on a file and don't want `rustfmt` to format it, include a `#[rustfmt::skip]` attribute at the top to prevent accidental large formatting commits.

8. **Rust in general**: Our goal is to be reasonably fast and do things in a performant manner, but the focus will be on real-world performance.  Things being theoretically safe or being idiomatic rust is not a big concern. Things should be elegant and work well.

#### TypeScript Type Safety

9. **Strict Type Enforcement in TypeScript**: Avoid the use of "any" or non-specific types. All data should strictly adhere to well-defined types to minimize the risk of introducing runtime errors due to type mismatches.
